### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by AndreiMotinga**

### CODE REVIEW

File: solidus_globalize.gemspec [hunk 1]
```diff
@@ -24,6 +24,7 @@ Gem::Specification.new do |s|
 
   s.add_runtime_dependency 'friendly_id-globalize'
   s.add_runtime_dependency 'globalize', '~> 5.0'
+  s.add_runtime_dependency 'solidus_auth_devise', '~> 1.6'
```
Comment (tvdeyen) [inline solidus_globalize.gemspec:27]: `solidus_globalize` does not need `solidus_auth_devise` to run. The dummy app needs it to have authenticated user, but it's not necessary for all stores. Lots of stores may have there own authentication user. With adding this to the `gemspec` you force all users of `solidus_globalize` to also use `solidus_auth_devise`.

File: solidus_globalize.gemspec [hunk 2]
```diff
@@ -38,9 +39,12 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'pry-rails', '>= 0.3.0'
   s.add_development_dependency 'rubocop', '>= 0.24.1'
   s.add_development_dependency 'rspec-rails', '~> 3.1'
+  s.add_development_dependency 'spring'
```
Comment (tvdeyen) [inline solidus_globalize.gemspec:42]: Please don't add gems like `spring` into the `gemspec`. All of these files will get installed together with the gem. If you want to have gems for local development please use the `Gemfile` for it.

File: spec/controllers/spree/products_controller_spec.rb [hunk 3]
```diff
@@ -8,7 +8,7 @@ module Spree
 
     context "visit non translated product page via permalink on url" do
       it "displays pages successfully" do
-        spree_get :show, id: product.slug, locale: 'pt-BR'
+        get :show, params: { id: product.slug, locale: 'pt-BR' }
```
Comment (tvdeyen) [inline spec/controllers/spree/products_controller_spec.rb:11]: In order to get this working with older versions of Solidus you also need to add the `rails_test_params_backport` gem to the `Gemfile`, like here:

https://github.com/solidusio/solidus_paypal_braintree/blob/master/Gemfile#L6-L11

**Review by tvdeyen — CHANGES_REQUESTED** [review]
### Thanks for your pull request

And looking into this gem. This is very much appreciated :cake:

#### I have some comments about how we like to have PRs in Solidus:

1. Please make your PR as small as possible, so it can be easily reviewed.

  For instance, this PR could be split up in three PRs. One for the test suite fixes (should always be the first), then tackling depression warnings and then one for adding new features, like translations and such. This really helps the maintainers to review and merge a PR faster. Also build issues can be tracked down more easily.

2. The next thing that helps maintainers and reviewers are precise and helpful commit messages.

  Even if it seems totally obvious for you what and why you did a certain commit, other readers may not have the same focus like you had while writing that code. 
  Consider that most of the maintainers are doing this voluntary and do not have much time to think into what you are doing and need to be able to glinse very fast through the changes and make a decision based on what they see in the changeset. 
  In order to get a pull request merged faster this is very helpful.

  Think of writing a story for someone that does not have a clue about what you are doing here and try to explain this person what and why you did this. This is very important for open source to keep this in mind. Just grab a open source project and go through the history of a project and try to find out why someone made a certain decision. This is most of the time very hard. Why try to be better at this. One of the reasons, why we forked Spree :) A very good read about that topic: https://chris.beams.io/posts/git-commit/

3. Make changes that are necessary for everyone. Although it might be helpful for ypu to use spring it might not be for others. Maybe they prefer to use other solutions like zeus, etc.

Not all of the above mentioned things apply to you and your PR, though. Again, **your PR is very much appreciated.**

See this as good advice for future contributors and even **potential new maintainers** of this gem.

**Comment by AndreiMotinga** [comment]
@tvdeyen, thanks you for your detailed response, I appreciate it. 

just to clarify, do you want this particular pr broken up? 

also, checks are failing on travis for older versions of solidus, what do you expect to be done about that, so that you could merge ?

**Comment by tvdeyen** [comment]
> just to clarify, do you want this particular pr broken up?

Yes. The first one being the fixes for the test suite.

> also, checks are failing on travis for older versions of solidus, what do you expect to be done about that, so that you could merge ?

They need to be fixed first. We need a green test suite.

**Pull request closed without merging**